Update global configs for v 1.1 (#1319)
dbt-labs/docs.getdbt.com: website/docs/reference/global-configs.md

</File>

<VersionBlock firstVersion="1.1">

### Cache database objects for selected resource

:::caution Experimental config flag
This should be considered an _unstable_ config flag because it is experimental and subject to change. We reserve the right to make breaking changes to this config flag.
:::

At the start of runs, dbt caches metadata about all the objects in all the schemas where it might materialize resources (such as models). By default, dbt caches all schemas related to the project. When this config is enabled, dbt will only cache schemas related to selected resources for the current run. This can offer significant speed improvements when running a small subset of a large project.

> For example, to improve speed and performance while focused on Salesforce models you would select the source and use the `cache-selected-only` flag:

Reviewer: This feels to me like an important nuance to capture, even if it makes the sentence clunky:
```suggestion
For example, to improve speed and performance while focused on developing Salesforce models, which are materialized into their own dedicated schema, you would select those models and pass the `cache-selected-only` flag:
```

```text

> $ dbt build --select source:salesforce 
> $ dbt --cache-selected-only run

Reviewer: Let's combine these into a single command:
```suggestion
$ dbt --cache-selected-only run --select salesforce
```
In order to work, that single command must include the config and the selection criteria, and it must be selecting buildable resources (models, not sources).

</File>

<VersionBlock firstVersion="1.1">

### Cache database objects for selected resource

:::caution Experimental config flag
This should be considered an _unstable_ config flag because it is experimental and subject to change. We reserve the right to make breaking changes to this config flag.
:::

> When true, dbt caches database objects relevant to selected resource. Otherwise, dbt will cache all database objects related to the project.

Reviewer: We're missing some more generic documentation about why caching exists and how it works, akin to the Notion doc I shared in a previous comment. In the meantime, it might be worth an extra bit of explanation here.
```suggestion
At the start of runs, dbt caches metadata about all the objects in all the schemas where it might materialize resources (such as models). By default, dbt caches all schemas related to the project. When this config is enabled, dbt will only cache schemas related to selected resources for the current run. This can offer significant speed improvements when running a small subset of a large project.
```

For example, to improve speed and performance while focused on Salesforce models you would select the source and use the `cache-selected-only` flag:

```text

$ dbt build --select source:salesforce 
$ dbt --cache-selected-only run

> ```

Reviewer: Meta-comment: dbt doesn't actually cache sources, only buildable resources (models, seeds, snapshots, tests if `--store-failures`). Again, we're missing broader explanatory resources about caching, which we can only make up for so much in this one little section.
```suggestion
For example, if your project materializes Salesforce models in a dedicated schema, and you want to focus on building those models, you can improve start-up speed by using the `--cache-selected-only` flag:

```text
$ dbt --cache-selected-only build --select salesforce
```
This will run fewer metadata queries upfront, and start building faster.

Author: Thanks for adding the context, @jtcohen6 I think adding caching content to the docs and linking to it will really help!

:::

</File>

### Partial Parsing

The `PARTIAL_PARSE` config can turn partial parsing on or off in your project. See [the docs on parsing](parsing#partial-parsing) for more details.

<File name='profiles.yml'>

```yaml

config:
  partial_parse: true

```

</File>

<File name='Usage'>

```text
dbt --no-partial-parse run
```

</File>

### Printer width

By default, dbt will print out lines padded to 80 characters wide. You can change this setting by adding the following to your `profiles.yml` file:

<File name='profiles.yml'>

```yaml
config:
  printer_width: 120
```

</File>

> ### Send anonymous usage stats

Reviewer: One update here: We just merged a community contribution that supports an env var named `DO_NOT_TRACK`, which will turn on/off the same config: https://github.com/dbt-labs/dbt-core/pull/5000

These are equivalent:
- `DO_NOT_TRACK=1` is the same as `DBT_SEND_ANONYMOUS_USAGE_STATS=False
- `DO_NOT_TRACK=2` is the same as `DBT_SEND_ANONYMOUS_USAGE_STATS=True`

Author: Added this in https://github.com/dbt-labs/docs.getdbt.com/pull/1319/commits/af4adbe7e555faf316fb6b9fa174a8419d18bc5e

</File>

### Strict

As of v1.0, the `-S` or `--strict` flag has been deprecated.

<VersionBlock firstVersion="1.1">
### Suppress non-error logs in output

> By default, dbt shows all logs in standard out (stdout). You can use the `QUIET` config to show only error logs in stdout. For example, you might narrow down your logs to error logs only because it might help you more easily find and debug a jinja error.

Reviewer: The reason someone would use `QUIET` is to:
- suppress all logs except errors
- still include the output of anything passed to the `print()` macro (#1320)

Reviewer: A community contributor also just added a `NO_PRINT` config / `--no-print` flag: https://github.com/dbt-labs/dbt-core/pull/4854

It would make sense to document these two alongside each other, or link between them.

Author: Addressed these in https://github.com/dbt-labs/docs.getdbt.com/pull/1319/commits/af4adbe7e555faf316fb6b9fa174a8419d18bc5e

  send_anonymous_usage_stats: False
```

You can also use the DO_NOT_TRACK environmental variable to enable or disable sending anonymous data. For more information, see [Environmental variables](/dbt-cloud/using-dbt-cloud/cloud-environment-variables).

`DO_NOT_TRACK=1` is the same as `DBT_SEND_ANONYMOUS_USAGE_STATS=False`

> `DO_NOT_TRACK=2` is the same as `DBT_SEND_ANONYMOUS_USAGE_STATS=True`

Reviewer: Sorry, I may have confused this one. The two are both boolean and exactly inverses; `DO_NOT_TRACK` just uses `0` instead of `False` by convention.
```suggestion
`DO_NOT_TRACK=0` is the same as `DBT_SEND_ANONYMOUS_USAGE_STATS=True`
```

As of v1.0, the `-S` or `--strict` flag has been deprecated.

<VersionBlock firstVersion="1.1">

### Suppress non-error logs in output

> By default, dbt shows all logs in standard out (stdout). You can use the `QUIET` config to show only error logs in stdout. Logs will still include the output of anything passed to the `print()` macro.  For example, you might suppress all but error logs to more easily find and debug a jinja error.

Reviewer: 👍
